src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    const currentAnimal = Animal.alive.find(animal => animal === herbivore);
+
+    if (currentAnimal.hidden !== undefined && currentAnimal.hidden !== true) {
+      currentAnimal.health -= 50;
+    }
```

Comment:
This action should happen only if `hidden` is true, actually there is no need to check if it's not undefined or not true, simplify this condition. Also to check if this object is an instance of `Herbivore` better to use `instanceof`

Reply:
Oh, no, I was wrong here, sorry, just check if this object is instance of `Herbivore` and if `hidden` is `false` here

Reply:
done)